### PR TITLE
Add Bot#parse_mentions

### PR DESCRIPTION
**Pull request by SanksTheYokai**
Add `Bot#parse_mentions` to support the parsing of multiple mentions

### CODE REVIEW

**Comment by z64** [comment]
The git history for this PR is not correct; please delete the 3 unrelated commits

**Comment by SanksTheYokai** [comment]
Fixed _!_

File: lib/discordrb/bot.rb [hunk 1]
```diff
@@ -426,28 +426,41 @@ def update_oauth_application(name, redirect_uris, description = '', icon = nil)
...
+    # Gets the users, channels, roles and emoji from a string.
+    # @param mentions [String] The mentions, which should look like `<@12314873129>`, `<#123456789>`, `<@&123456789>` or `<:name:126328:>`.
+    # @param server [Server, nil] The server of the associated mentions. (recommended for role parsing, to speed things up)
+    # @return [Array<User, Channel, Role, Emoji>] The array of users, channels, roles and emoji identified by the mentions, or `nil` if none exists.
+    def parse_mentions(mentions, server = nil)
+      array_to_return = []
+      while mentions.include?('<') && mentions.include?('>')
+        mentions = mentions.split('<', 2)[1]
+        next unless mentions.split('>', 2).first.length < mentions.split('<', 2).first.length
+        mention = mentions.split('>', 2).first
```
Comment (z64) [inline lib/discordrb/bot.rb]: My main concern with this diff is the above 4 lines. While it works to our specs, this code is not maintainable; there's no sign of semantics and it's impossible to understand what it is doing in your head. For that same reason, I don't know what to suggest to improve it, other than it must be made more clear what this code is doing. I'm unsure about all the string splitting that is going on as well.

**Comment by z64** [comment]
Hi @SanksTheYokai !

With #487 we have updated Rubocop. Could you rebase this on the current master and re-run rubocop to pick up any additional changes necessary & resolve the (simple) conflict?

**Pull request merged**